jmx_prometheus_javaagent/src/main/java/io/prometheus/jmx/JavaAgent.java

```diff
@@ -1,24 +1,121 @@
 package io.prometheus.jmx;
 
+import io.prometheus.client.CollectorRegistry;
 import io.prometheus.client.exporter.MetricsServlet;
 import io.prometheus.client.hotspot.DefaultExports;
+
+import java.io.File;
+import java.io.IOException;
 import java.lang.instrument.Instrumentation;
 import java.io.FileReader;
+import java.nio.file.FileSystem;
+import java.nio.file.Path;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
 import org.eclipse.jetty.server.Server;
 import org.eclipse.jetty.servlet.ServletContextHandler;
 import org.eclipse.jetty.servlet.ServletHolder;
 import org.eclipse.jetty.util.thread.QueuedThreadPool;
 
+import javax.management.MalformedObjectNameException;
+
 public class JavaAgent {
    static Server server;
 
+   static JmxCollector previousJmxCollector = null;
+
+  /**
+   * This will load the config located at @configPath and replace the previous one.
+   * @param configPath
+   * @throws IOException
+   */
+   public static void loadConfig(String configPath) throws IOException {
+     if(previousJmxCollector != null) {
+       CollectorRegistry.defaultRegistry.unregister(previousJmxCollector);
```

Comment:
There's a race here where no results may be returned. It'd simpler to have the Collector check the timestamp on each collection, and reload if needed.